src/gui/gui.cpp

```diff
@@ -5680,6 +5705,42 @@ bool FurnaceGUI::loop() {
               }
               break;
             }
+            case GUI_FILE_EXPORT_MML: {
+              SafeWriter* w = nullptr;
+
+              switch (mmlExportType) {
+                case MML_EXPORT_MMLGB: // mmlgb
+                  w = e->saveMMLGB(mmlExportUseLegacyNoise);
+                  break;
+                case MML_EXPORT_AMK: // AddMusicK
+                  w = e->saveMMLSNESAMK(mmlExportAMKVersion);
+                  break;
+                default:
+                  showError(_("This MML format is currently unimplemented."));
+                  break;
+              }
+
+              if (w != nullptr) {
+                FILE* f = ps_fopen(copyOfName.c_str(), "wb");
+                if (f != nullptr) {
+                  fwrite(w->getFinalBuf(), 1, w->size(), f);
```

Comment:
Check return value - it is important.
It must be equal to w->size() (if it is not, then we ran out of space or a write error occurred).